From 42d3c03b11a98f2f2a6c461c8858d564ed342d55 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Fri, 27 May 2022 13:10:33 +0300
Subject: [PATCH 18/42] fdts: s32*: unify SIUL2 modules

Unify the two SIUL2 modules inside the dts files.

Issue: ALB-8868

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 fdts/s32g.dtsi             | 221 +++++++++++++++++++------------------
 fdts/s32g274a-bluebox3.dts | 136 +++++++++--------------
 fdts/s32g274a-emu.dts      |  22 ++--
 fdts/s32g399a-emu.dts      |  24 ++--
 fdts/s32gxxxa-evb.dtsi     | 156 ++++++++++----------------
 fdts/s32gxxxa-rdb.dtsi     | 146 ++++++++++--------------
 fdts/s32r45-evb.dts        | 118 ++++++++------------
 fdts/s32r45.dtsi           | 144 ++++++++++++++----------
 8 files changed, 431 insertions(+), 536 deletions(-)

diff --git a/fdts/s32g.dtsi b/fdts/s32g.dtsi
index 43de288c3..87d7f12d8 100644
--- a/fdts/s32g.dtsi
+++ b/fdts/s32g.dtsi
@@ -6,6 +6,7 @@
 #include <dt-bindings/clock/s32g-scmi-clock.h>
 #include <dt-bindings/nvmem/s32cc-siul2-nvmem.h>
 #include <dt-bindings/pinctrl/s32g-pinctrl.h>
+#include <dt-bindings/memory/s32-siul2.h>
 
 #include "s32cc.dtsi"
 / {
@@ -31,34 +32,93 @@
 	};
 
 	soc {
-		siul2_0: siul2_0@4009c000 {
+		siul2@4009c000 {
 			compatible = "simple-mfd";
 			status = "okay";
-			reg = <0x0 0x4009c000 0x0 0x2000>;
 			u-boot,dm-pre-reloc;
-
-			pinctrl0: siul2-pinctrl0 {
-				compatible = "nxp,s32cc-siul2-pinctrl";
+			#address-cells = <2>;
+			#size-cells = <2>;
+
+				 /* MIDR */
+			ranges = <MIDR_SIUL2_0 0x0 0x0 0x4009c000 0x0 0x10>,
+				 <MIDR_SIUL2_1 0x0 0x0 0x44010000 0x0 0x10>,
+
+				 /* MSCR 0-101 */
+				 <MSCRS_SIUL2_0 0x0 0x0 0x4009c240 0x0 0x198>,
+				 /* MSCR 112-122 */
+				 <MSCRS_SIUL2_1_0 0x0 0x0 0x44010400 0x0 0x2c>,
+				 /* MSCR 144-190 */
+				 <MSCRS_SIUL2_1_1 0x0 0x0 0x44010480 0x0 0xbc>,
+
+				 /* IMCR range 512–595 */
+				 <IMCRS_SIUL2_0 0x0 0x0 0x4009ca40 0x0 0x150>,
+				 /* IMCR range 631 - 909 */
+				 <IMCRS_SIUL2_1_0 0x0 0x0 0x44010c1c 0x0 0x45c>,
+				 /* IMCR range 910 - 941 (EIRQs) */
+				 <IMCRS_SIUL2_1_EIRQS 0x0 0x0 0x44011078 0x0 0x80>,
+				 /* IMCR range 942 - 1007 */
+				 <IMCRS_SIUL2_1_1 0x0 0x0 0x440110f8 0x0 0x108>,
+
+				 /* Output pads */
+				 <OPADS_SIUL2_0 0x0 0x0 0x4009d700 0x0 0x10>,
+				 <OPADS_SIUL2_1 0x0 0x0 0x44011700 0x0 0x18>,
+
+				 /* Input pads */
+				 <IPADS_SIUL2_0 0x0 0x0 0x4009d740 0x0 0x10>,
+				 <IPADS_SIUL2_1 0x0 0x0 0x44011740 0x0 0x18>,
+
+				 /* EIRQs */
+				 <EIRQS_SIUL2_1 0x0 0x0 0x44010010 0x0 0xb4>;
+
+			pinctrl: siul2-pinctrl@4009c240 {
+				compatible = "nxp,s32g-siul2-pinctrl";
 				#pinctrl-cells = <2>;
+
+				reg = <MSCRS_SIUL2_0    0 0x0 0x198>,
+				      <MSCRS_SIUL2_1_0  0 0x0 0x2c>,
+				      <MSCRS_SIUL2_1_1  0 0x0 0xbc>,
+				      <IMCRS_SIUL2_0    0 0x0 0x150>,
+				      <IMCRS_SIUL2_1_0  0 0x0 0x45c>,
+				      <IMCRS_SIUL2_1_1  0 0x0 0x108>;
+
 				/* MSCR range */
-				pins = <&pinctrl0 0 101>,
+				nxp,pins = <0   101
+					    112 122
+					    144 190
 				/* IMCR range */
-				<&pinctrl0 512 595>;
+					    512 595
+					    631 909
+					    942 1007>;
 				status = "okay";
 			};
 
-			gpio0: siul2-gpio0 {
+			gpio: siul2-gpio@4009d700 {
 				compatible = "nxp,s32cc-siul2-gpio";
+
+				reg = <OPADS_SIUL2_0        0 0x0 0x10>,
+				      <OPADS_SIUL2_1        0 0x0 0x18>,
+				      <IPADS_SIUL2_0        0 0x0 0x10>,
+				      <IPADS_SIUL2_1        0 0x0 0x18>,
+				      <EIRQS_SIUL2_1        0 0x0 0xb4>,
+				      <IMCRS_SIUL2_1_EIRQS  0 0x0 0x80>;
+				reg-names = "opads0", "opads1", "ipads0",
+					    "ipads1", "eirqs", "eirq-imcrs";
+
 				#gpio-cells = <2>;
 				gpio-controller;
-				gpio-ranges = <&pinctrl0 0 0 102>;
+					/* GPIO 0-101 */
+				gpio-ranges = <&pinctrl 0   0   102>,
+					/* GPIO 112-190 */
+					      <&pinctrl 112 112 79>;
 				status = "okay";
 			};
 
 			siul2_0_nvram: siul2_0_nvram {
+				reg = <MIDR_SIUL2_0 0 0x0 0x10>;
 				#address-cells = <1>;
 				#size-cells = <1>;
 
+				compatible = "nxp,s32g-siul2_0-nvram";
 				status = "okay";
 				u-boot,dm-pre-reloc;
 
@@ -103,37 +163,13 @@
 					    S32CC_CELL_SIZE>;
 				};
 			};
-		};
-
-		siul2_1@44010000 {
-			compatible = "simple-mfd";
-			status = "okay";
-			reg = <0x0 0x44010000 0x0 0x2000>;
-			u-boot,dm-pre-reloc;
-
-			pinctrl1: siul2-pinctrl1 {
-				compatible = "nxp,s32cc-siul2-pinctrl";
-				#pinctrl-cells = <2>;
-				/* MSCR range */
-				pins = <&pinctrl1 112 190>,
-				/* IMCR range */
-				<&pinctrl1 631 1023>;
-				status = "okay";
-			};
-
-			gpio1: siul2-gpio1 {
-				compatible = "nxp,s32cc-siul2-gpio";
-				#gpio-cells = <2>;
-				gpio-controller;
-				gpio-ranges = <&pinctrl1 0 112 11>,
-					<&pinctrl1 11 144 47>;
-				status = "okay";
-			};
 
 			siul2_1_nvram: siul2_1_nvram {
+				reg = <MIDR_SIUL2_1 0 0x0 0x10>;
 				#address-cells = <1>;
 				#size-cells = <1>;
 
+				compatible = "nxp,s32g-siul2_1-nvram";
 				status = "okay";
 				u-boot,dm-pre-reloc;
 
@@ -190,15 +226,12 @@
 				"mac1_rx_rgmii", "mac1_tx_rgmii",
 				"mac2_rx_sgmii", "mac2_tx_sgmii",
 				"mac2_rx_rgmii", "mac2_tx_rgmii";
-			pinctrl-0 = <&pinctrl0_pfe0 &pinctrl0_pfe0_mdio &pinctrl1_pfe0
-				&pinctrl1_pfe0_mdio>;
-			pinctrl-1 = <&pinctrl0_pfe0_mdio &pinctrl1_pfe0_mdio>;
-			pinctrl-2 = <&pinctrl0_pfe1 &pinctrl0_pfe1_mdio &pinctrl1_pfe1
-				&pinctrl1_pfe1_mdio>;
-			pinctrl-3 = <&pinctrl0_pfe1_mdio &pinctrl1_pfe1_mdio>;
-			pinctrl-4 = <&pinctrl0_pfe2 &pinctrl0_pfe2_mdio &pinctrl1_pfe2
-				&pinctrl1_pfe2_mdio>;
-			pinctrl-5 = <&pinctrl0_pfe2_mdio &pinctrl1_pfe2_mdio>;
+			pinctrl-0 = <&pinctrl_pfe0 &pinctrl_pfe0_mdio>;
+			pinctrl-1 = <&pinctrl_pfe0_mdio &pinctrl_pfe0_mdio>;
+			pinctrl-2 = <&pinctrl_pfe1 &pinctrl_pfe1_mdio>;
+			pinctrl-3 = <&pinctrl_pfe1_mdio &pinctrl_pfe1_mdio>;
+			pinctrl-4 = <&pinctrl_pfe2 &pinctrl_pfe2_mdio>;
+			pinctrl-5 = <&pinctrl_pfe2_mdio>;
 			pinctrl-names = "pfe0_rgmii", "pfe0_sgmii",
 				"pfe1_rgmii", "pfe1_sgmii",
 				"pfe2_rgmii", "pfe2_sgmii";
@@ -272,8 +305,8 @@
 };
 
 &gmac0 {
-	pinctrl-0 = <&pinctrl0_gmac0 &pinctrl0_gmac0_mdio>;
-	pinctrl-1 = <&pinctrl0_gmac0_mdio>;
+	pinctrl-0 = <&pinctrl_gmac0 &pinctrl_gmac0_mdio>;
+	pinctrl-1 = <&pinctrl_gmac0_mdio>;
 	pinctrl-names = "gmac_rgmii", "gmac_sgmii";
 };
 
@@ -302,10 +335,10 @@
 	status = "okay";
 };
 
-&pinctrl0 {
-	board_generic_pinctrl0 {
+&pinctrl {
+	board_generic_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_sd0: pinctrl0_sd0 {
+		pinctrl_sd0: pinctrl_sd0 {
 			fsl,pins = <PC14_MSCR_S32G PC14_SD0_CLK_CFG
 				    PC15_MSCR_S32G PC15_SD0_CMD_CFG
 				    PD00_MSCR_S32G PD00_SD0_D0_CFG
@@ -332,14 +365,14 @@
 				    >;
 		};
 
-		pinctrl0_gmac0_mdio: pinctrl0_gmac0_mdio {
+		pinctrl_gmac0_mdio: pinctrl_gmac0_mdio {
 			fsl,pins = <PD12_MSCR_S32G PD12_GMAC0_MDC_CFG
 				    PD13_MSCR_S32G PD13_GMAC0_MDIO_CFG
 				    GMAC0_MDIO_IMCR PD13_GMAC0_MDIO_IN
 				    >;
 		};
 
-		pinctrl0_gmac0: pinctrl0_gmac0 {
+		pinctrl_gmac0: pinctrl_gmac0 {
 			fsl,pins = <PE02_MSCR_S32G PE02_GMAC0_TX_CLK_CFG
 				    PE03_MSCR_S32G PE03_GMAC0_TX_EN_CFG
 				    PE04_MSCR_S32G PE04_GMAC0_TX_D0_CFG
@@ -362,58 +395,16 @@
 				    >;
 		};
 
-		pinctrl0_pfe0_mdio: pinctrl0_pfe0_mdio {
+		pinctrl_pfe0_mdio: pinctrl_pfe0_mdio {
 			fsl,pins = <PE15_MSCR_S32G PE15_PFE0_MDIO_CFG
 				    PF02_MSCR_S32G PF02_PFE0_MDC_CFG
+				    PFE0_MDIO_IMCR PE15_PFE0_MDIO_IN
 				    >;
 		};
 
-		pinctrl0_pfe0: pinctrl0_pfe0 {
-			fsl,pins = <PE14_MSCR_S32G PE14_PFE0_TX_EN_CFG>;
-		};
-
-		pinctrl0_pfe1_mdio: pinctrl0_pfe1_mdio {
-			fsl,pins = <PD12_MSCR_S32G PD12_PFE1_MDC_CFG
-				    PD13_MSCR_S32G PD13_PFE1_MDIO_CFG
-				    >;
-		};
-
-		pinctrl0_pfe1: pinctrl0_pfe1 {
-			fsl,pins = <PE02_MSCR_S32G PE02_PFE1_TX_CLK_CFG
-				    PE03_MSCR_S32G PE03_PFE1_TX_EN_CFG
-				    PE04_MSCR_S32G PE04_PFE1_TX_D0_CFG
-				    PE05_MSCR_S32G PE05_PFE1_TX_D1_CFG
-				    PE06_MSCR_S32G PE06_PFE1_TX_D2_CFG
-				    PE07_MSCR_S32G PE07_PFE1_TX_D3_CFG
-				    PE08_MSCR_S32G PE08_PFE1_RX_CLK_CFG
-				    PE09_MSCR_S32G PE09_PFE1_RX_DV_CFG
-				    PE10_MSCR_S32G PE10_PFE1_RX_D0_CFG
-				    PE11_MSCR_S32G PE11_PFE1_RX_D1_CFG
-				    PE12_MSCR_S32G PE12_PFE1_RX_D2_CFG
-				    PE13_MSCR_S32G PE13_PFE1_RX_D3_CFG
-				    >;
-		};
-
-		pinctrl0_pfe2_mdio: pinctrl0_pfe2_mdio {
-			fsl,pins = <PE15_MSCR_S32G PE15_PFE2_MDIO_CFG
-				    PF02_MSCR_S32G PF02_PFE2_MDC_CFG
-				    >;
-		};
-
-		pinctrl0_pfe2: pinctrl0_pfe2 {
-			fsl,pins = <PE14_MSCR_S32G PE14_PFE2_TX_EN_CFG>;
-		};
-	};
-};
-
-&pinctrl1 {
-	board_generic_pinctrl1 {
-		pinctrl1_pfe0_mdio: pinctrl1_pfe0_mdio {
-			fsl,pins = <PFE0_MDIO_IMCR PE15_PFE0_MDIO_IN>;
-		};
-
-		pinctrl1_pfe0: pinctrl1_pfe0 {
-			fsl,pins = <PH01_MSCR_S32G PH01_PFE0_TX_D1_CFG
+		pinctrl_pfe0: pinctrl_pfe0 {
+			fsl,pins = <PE14_MSCR_S32G PE14_PFE0_TX_EN_CFG
+				    PH01_MSCR_S32G PH01_PFE0_TX_D1_CFG
 				    PH02_MSCR_S32G PH02_PFE0_TX_D2_CFG
 				    PH03_MSCR_S32G PH03_PFE0_TX_D3_CFG
 				    PH04_MSCR_S32G PH04_PFE0_RX_CLK_CFG
@@ -433,12 +424,27 @@
 				    >;
 		};
 
-		pinctrl1_pfe1_mdio: pinctrl1_pfe1_mdio {
-			fsl,pins = <PFE1_MDIO_IMCR PD13_PFE1_MDIO_IN>;
+		pinctrl_pfe1_mdio: pinctrl_pfe1_mdio {
+			fsl,pins = <PD12_MSCR_S32G PD12_PFE1_MDC_CFG
+				    PD13_MSCR_S32G PD13_PFE1_MDIO_CFG
+				    PFE1_MDIO_IMCR PD13_PFE1_MDIO_IN
+				    >;
 		};
 
-		pinctrl1_pfe1: pinctrl1_pfe1 {
-			fsl,pins = <PFE1_RX_CLK_IMCR PE08_PFE1_RX_CLK_IN
+		pinctrl_pfe1: pinctrl_pfe1 {
+			fsl,pins = <PE02_MSCR_S32G PE02_PFE1_TX_CLK_CFG
+				    PE03_MSCR_S32G PE03_PFE1_TX_EN_CFG
+				    PE04_MSCR_S32G PE04_PFE1_TX_D0_CFG
+				    PE05_MSCR_S32G PE05_PFE1_TX_D1_CFG
+				    PE06_MSCR_S32G PE06_PFE1_TX_D2_CFG
+				    PE07_MSCR_S32G PE07_PFE1_TX_D3_CFG
+				    PE08_MSCR_S32G PE08_PFE1_RX_CLK_CFG
+				    PE09_MSCR_S32G PE09_PFE1_RX_DV_CFG
+				    PE10_MSCR_S32G PE10_PFE1_RX_D0_CFG
+				    PE11_MSCR_S32G PE11_PFE1_RX_D1_CFG
+				    PE12_MSCR_S32G PE12_PFE1_RX_D2_CFG
+				    PE13_MSCR_S32G PE13_PFE1_RX_D3_CFG
+				    PFE1_RX_CLK_IMCR PE08_PFE1_RX_CLK_IN
 				    PFE1_RX_DV_IMCR PE09_PFE1_RX_DV_IN
 				    PFE1_RX_D0_IMCR PE10_PFE1_RX_D0_IN
 				    PFE1_RX_D1_IMCR PE11_PFE1_RX_D1_IN
@@ -447,13 +453,16 @@
 				    >;
 		};
 
-		pinctrl1_pfe2_mdio: pinctrl1_pfe2_mdio {
-			fsl,pins = <PFE2_MDIO_IMCR PE15_PFE2_MDIO_IN
+		pinctrl_pfe2_mdio: pinctrl_pfe2_mdio {
+			fsl,pins = <PE15_MSCR_S32G PE15_PFE2_MDIO_CFG
+				    PF02_MSCR_S32G PF02_PFE2_MDC_CFG
+				    PFE2_MDIO_IMCR PE15_PFE2_MDIO_IN
 				    >;
 		};
 
-		pinctrl1_pfe2: pinctrl1_pfe2 {
-			fsl,pins = <PH01_MSCR_S32G PH01_PFE2_TX_D1_CFG
+		pinctrl_pfe2: pinctrl_pfe2 {
+			fsl,pins = <PE14_MSCR_S32G PE14_PFE2_TX_EN_CFG
+				    PH01_MSCR_S32G PH01_PFE2_TX_D1_CFG
 				    PH02_MSCR_S32G PH02_PFE2_TX_D2_CFG
 				    PH03_MSCR_S32G PH03_PFE2_TX_D3_CFG
 				    PH04_MSCR_S32G PH04_PFE2_RX_CLK_CFG
diff --git a/fdts/s32g274a-bluebox3.dts b/fdts/s32g274a-bluebox3.dts
index f16fa756f..9354a1073 100644
--- a/fdts/s32g274a-bluebox3.dts
+++ b/fdts/s32g274a-bluebox3.dts
@@ -12,20 +12,20 @@
 };
 
 &spi0 {
-	pinctrl-0 = <&pinctrl0_dspi0 &pinctrl1_dspi0>;
+	pinctrl-0 = <&pinctrl_dspi0>;
 	pinctrl-names = "default";
 	status = "okay";
 };
 
 &spi1 {
-	pinctrl-0 = <&pinctrl0_dspi1 &pinctrl1_dspi1>;
+	pinctrl-0 = <&pinctrl_dspi1>;
 	pinctrl-names = "default";
 	status = "okay";
 };
 
 &usdhc0 {
 	/* By default, sd0 pins are able to work at 100Mhz and 200Mhz */
-	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-0 = <&pinctrl_sd0>;
 	pinctrl-1 = <>;
 	pinctrl-2 = <>;
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
@@ -41,7 +41,7 @@
 	status = "disabled";
 };
 
-&gpio0 {
+&gpio {
 	PA12 {
 		gpio-hog;
 		gpios = <12 0>;
@@ -74,10 +74,10 @@
 &i2c0 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
-	scl-gpios = <&gpio0 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c0>;
+	pinctrl-1 = <&pinctrl_i2c0_gpio>;
+	scl-gpios = <&gpio 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 
 	eeprom@50 {
@@ -100,30 +100,30 @@
 &i2c1 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
-	pinctrl-1 = <&pinctrl0_i2c1_gpio &pinctrl1_i2c1_gpio>;
-	scl-gpios = <&gpio0 19 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 20 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c1>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio 19 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 20 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 &i2c2 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
-	pinctrl-1 = <&pinctrl0_i2c2_gpio &pinctrl1_i2c2_gpio>;
-	scl-gpios = <&gpio0 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c2>;
+	pinctrl-1 = <&pinctrl_i2c2_gpio>;
+	scl-gpios = <&gpio 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 &i2c4 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
-	pinctrl-1 = <&pinctrl0_i2c4_gpio &pinctrl1_i2c4_gpio>;
-	scl-gpios = <&gpio0 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c4>;
+	pinctrl-1 = <&pinctrl_i2c4_gpio>;
+	scl-gpios = <&gpio 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 
 	vr5510@20 {
@@ -139,28 +139,30 @@
 	};
 };
 
-&pinctrl0 {
-	board_pinctrl0 {
+&pinctrl {
+	board_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_dspi0: pinctrl0_dspi0 {
+		pinctrl_dspi0: pinctrl_dspi0 {
 			fsl,pins = <PA13_MSCR_S32G PA13_SPI0_SCK_CFG
 				    PA14_MSCR_S32G PA14_SPI0_SIN_CFG
 				    PA15_MSCR_S32G PA15_SPI0_SOUT_CFG
 				    PB09_MSCR_S32G PB09_SPI0_CS1_CFG
 				    PB10_MSCR_S32G PB10_SPI0_CS2_CFG
+				    DSPI0_SIN_IMCR PA14_SPI0_SIN_IN
 				    >;
 		};
 
-		pinctrl0_dspi1: pinctrl0_dspi1 {
+		pinctrl_dspi1: pinctrl_dspi1 {
 			fsl,pins = <PA06_MSCR_S32G PA06_SPI1_SOUT_CFG
 				    PA07_MSCR_S32G PA07_SPI1_CS0_CFG
 				    PA08_MSCR_S32G PA08_SPI1_SCK_CFG
 				    PE04_MSCR_S32G PE04_SPI1_CS3_CFG
 				    PF15_MSCR_S32G PF15_SPI1_SIN_CFG
+				    DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
 				    >;
 		};
 
-		pinctrl0_i2c0: pinctrl0_i2c0 {
+		pinctrl_i2c0: pinctrl_i2c0 {
 			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_CFG
 				    PB01_MSCR_S32G PB01_I2C0_SCL_CFG
 				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
@@ -168,7 +170,7 @@
 				    >;
 		};
 
-		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+		pinctrl_i2c0_gpio: pinctrl_i2c0_gpio {
 			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_GPIO
 				    PB01_MSCR_S32G PB01_I2C0_SCL_GPIO
 				    I2C0_SDA_IMCR IMCR_DISABLED
@@ -176,43 +178,55 @@
 				    >;
 		};
 
-		pinctrl0_i2c1: pinctrl0_i2c1 {
+		pinctrl_i2c1: pinctrl_i2c1 {
 			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_CFG
 				    PB04_MSCR_S32G PB04_I2C1_SDA_CFG
+				    I2C1_SCL_IMCR PB03_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
 				    >;
 		};
 
-		pinctrl0_i2c1_gpio: pinctrl0_i2c1_gpio {
+		pinctrl_i2c1_gpio: pinctrl_i2c1_gpio {
 			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_GPIO
 				    PB04_MSCR_S32G PB04_I2C1_SDA_GPIO
+				    I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_i2c2: pinctrl0_i2c2 {
+		pinctrl_i2c2: pinctrl_i2c2 {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
 				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
+				    I2C2_SCL_IMCR PB05_I2C2_SCL_IN
+				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
 				    >;
 		};
 
-		pinctrl0_i2c2_gpio: pinctrl0_i2c2_gpio {
+		pinctrl_i2c2_gpio: pinctrl_i2c2_gpio {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
 				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
+				    I2C2_SCL_IMCR IMCR_DISABLED
+				    I2C2_SDA_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_i2c4: pinctrl0_i2c4 {
+		pinctrl_i2c4: pinctrl_i2c4 {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
 				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
+				    I2C4_SDA_IMCR PC01_I2C4_SDA_IN
+				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
 				    >;
 		};
 
-		pinctrl0_i2c4_gpio: pinctrl0_i2c4_gpio {
+		pinctrl_i2c4_gpio: pinctrl_i2c4_gpio {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
 				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
+				    I2C4_SDA_IMCR IMCR_DISABLED
+				    I2C4_SCL_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_qspi: pinctrl0_qspi {
+		pinctrl_qspi: pinctrl_qspi {
 			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
 				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
@@ -240,7 +254,7 @@
 				    >;
 		};
 
-		pinctrl0_uart0: pinctrl0_uart0 {
+		pinctrl_uart0: pinctrl_uart0 {
 			u-boot,dm-pre-reloc;
 			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
 				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
@@ -249,58 +263,8 @@
 	};
 };
 
-&pinctrl1 {
-	board_pinctrl1 {
-		pinctrl1_dspi0: pinctrl1_dspi0 {
-			fsl,pins = <DSPI0_SIN_IMCR PA14_SPI0_SIN_IN
-				    >;
-		};
-
-		pinctrl1_dspi1: pinctrl1_dspi1 {
-			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
-				    >;
-		};
-
-		pinctrl1_i2c1: pinctrl1_i2c1 {
-			fsl,pins = <I2C1_SCL_IMCR PB03_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
-			fsl,pins = <I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_i2c2: pinctrl1_i2c2 {
-			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
-				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c2_gpio: pinctrl1_i2c2_gpio {
-			fsl,pins = <I2C2_SCL_IMCR IMCR_DISABLED
-				    I2C2_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_i2c4: pinctrl1_i2c4 {
-			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
-				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
-				    >;
-		};
-
-		pinctrl1_i2c4_gpio: pinctrl1_i2c4_gpio {
-			fsl,pins = <I2C4_SDA_IMCR IMCR_DISABLED
-				    I2C4_SCL_IMCR IMCR_DISABLED
-				    >;
-		};
-	};
-};
-
 &qspi {
-	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-0 = <&pinctrl_qspi>;
 	pinctrl-names = "default";
 	status = "okay";
 
@@ -318,7 +282,7 @@
 };
 
 &uart0 {
-	pinctrl-0 = <&pinctrl0_uart0>;
+	pinctrl-0 = <&pinctrl_uart0>;
 	pinctrl-names = "default";
 	status = "okay";
 };
diff --git a/fdts/s32g274a-emu.dts b/fdts/s32g274a-emu.dts
index 6c2365e29..ffaeaa8ec 100644
--- a/fdts/s32g274a-emu.dts
+++ b/fdts/s32g274a-emu.dts
@@ -15,29 +15,21 @@
 	};
 };
 
-&pinctrl0 {
-	board_pinctrl0 {
+&pinctrl {
+	board_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_uart1: pinctrl0_uart1 {
+		pinctrl_uart1: pinctrl_uart1 {
 			u-boot,dm-pre-reloc;
 			fsl,pins = <PB09_MSCR_S32G PB09_LIN1_TX_CFG
-				    PB10_MSCR_S32G PB10_LIN1_RX_CFG>;
-		};
-	};
-};
-
-&pinctrl1 {
-	board_pinctrl1 {
-		u-boot,dm-pre-reloc;
-		pinctrl1_uart1: pinctrl1_uart1 {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <LIN1_RX_IMCR PB10_LIN1_RX_IN>;
+				    PB10_MSCR_S32G PB10_LIN1_RX_CFG
+				    LIN1_RX_IMCR PB10_LIN1_RX_IN
+				    >;
 		};
 	};
 };
 
 &uart1{
-	pinctrl-0 = <&pinctrl0_uart1 &pinctrl1_uart1>;
+	pinctrl-0 = <&pinctrl_uart1>;
 	pinctrl-names = "default";
 	status = "okay";
 };
diff --git a/fdts/s32g399a-emu.dts b/fdts/s32g399a-emu.dts
index d771ea5e9..6b6f78990 100644
--- a/fdts/s32g399a-emu.dts
+++ b/fdts/s32g399a-emu.dts
@@ -16,35 +16,27 @@
 	};
 };
 
-&pinctrl0 {
-	board_pinctrl0 {
+&pinctrl {
+	board_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_uart1: pinctrl0_uart1 {
+		pinctrl_uart1: pinctrl_uart1 {
 			u-boot,dm-pre-reloc;
 			fsl,pins = <PB09_MSCR_S32G PB09_LIN1_TX_CFG
-				    PB10_MSCR_S32G PB10_LIN1_RX_CFG>;
-		};
-	};
-};
-
-&pinctrl1 {
-	board_pinctrl1 {
-		u-boot,dm-pre-reloc;
-		pinctrl1_uart1: pinctrl1_uart1 {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <LIN1_RX_IMCR PB10_LIN1_RX_IN>;
+				    PB10_MSCR_S32G PB10_LIN1_RX_CFG
+				    LIN1_RX_IMCR PB10_LIN1_RX_IN
+				    >;
 		};
 	};
 };
 
 &uart1{
-	pinctrl-0 = <&pinctrl0_uart1 &pinctrl1_uart1>;
+	pinctrl-0 = <&pinctrl_uart1>;
 	pinctrl-names = "default";
 	status = "okay";
 };
 
 &usdhc0 {
-	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-0 = <&pinctrl_sd0>;
 	pinctrl-1 = <>;
 	pinctrl-2 = <>;
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
diff --git a/fdts/s32gxxxa-evb.dtsi b/fdts/s32gxxxa-evb.dtsi
index ba4985149..d63e01a1a 100644
--- a/fdts/s32gxxxa-evb.dtsi
+++ b/fdts/s32gxxxa-evb.dtsi
@@ -16,7 +16,7 @@
 		reg = <0x0 0x44064000 0x0 0x200>;
 		clocks = <&clks S32G_SCMI_CLK_USB_MEM>,
 			   <&clks S32G_SCMI_CLK_USB_LOW>;
-		pinctrl-0 = <&pinctrl0_usb &pinctrl1_usb>;
+		pinctrl-0 = <&pinctrl_usb>;
 		pinctrl-names = "default";
 		phys = <&saf1508_phy>;
 		dr_mode = "host";
@@ -30,7 +30,7 @@
 };
 
 &usdhc0 {
-	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-0 = <&pinctrl_sd0>;
 	pinctrl-1 = <>;
 	pinctrl-2 = <>;
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
@@ -38,13 +38,13 @@
 };
 
 &spi1 {
-	pinctrl-0 = <&pinctrl0_dspi1 &pinctrl1_dspi1>;
+	pinctrl-0 = <&pinctrl_dspi1>;
 	pinctrl-names = "default";
 	status = "okay";
 };
 
 &spi5 {
-	pinctrl-0 = <&pinctrl0_dspi5 &pinctrl1_dspi5>;
+	pinctrl-0 = <&pinctrl_dspi5>;
 	pinctrl-names = "default";
 	status = "okay";
 
@@ -95,10 +95,10 @@
 &i2c0 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
-	scl-gpios = <&gpio0 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c0>;
+	pinctrl-1 = <&pinctrl_i2c0_gpio>;
+	scl-gpios = <&gpio 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -106,10 +106,10 @@
 &i2c1 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
-	pinctrl-1 = <&pinctrl0_i2c1_gpio &pinctrl1_i2c1_gpio>;
-	scl-gpios = <&gpio0 19 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 20 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c1>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio 19 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 20 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -117,10 +117,10 @@
 &i2c2 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
-	pinctrl-1 = <&pinctrl0_i2c2_gpio &pinctrl1_i2c2_gpio>;
-	scl-gpios = <&gpio0 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c2>;
+	pinctrl-1 = <&pinctrl_i2c2_gpio>;
+	scl-gpios = <&gpio 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -128,10 +128,10 @@
 &i2c4 {
 	status = "okay";
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
-	pinctrl-1 = <&pinctrl0_i2c4_gpio &pinctrl1_i2c4_gpio>;
-	scl-gpios = <&gpio0 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c4>;
+	pinctrl-1 = <&pinctrl_i2c4_gpio>;
+	scl-gpios = <&gpio 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	clock-frequency = <100000>;
 
 	vr5510@20 {
@@ -147,10 +147,10 @@
 	};
 };
 
-&pinctrl0 {
-	board_pinctrl0 {
+&pinctrl {
+	board_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_i2c0: pinctrl0_i2c0 {
+		pinctrl_i2c0: pinctrl_i2c0 {
 			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_CFG
 				    PB01_MSCR_S32G PB01_I2C0_SCL_CFG
 				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
@@ -158,7 +158,7 @@
 				    >;
 		};
 
-		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+		pinctrl_i2c0_gpio: pinctrl_i2c0_gpio {
 			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_GPIO
 				    PB01_MSCR_S32G PB01_I2C0_SCL_GPIO
 				    I2C0_SDA_IMCR IMCR_DISABLED
@@ -166,43 +166,55 @@
 				    >;
 		};
 
-		pinctrl0_i2c1: pinctrl0_i2c1 {
+		pinctrl_i2c1: pinctrl_i2c1 {
 			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_CFG
 				    PB04_MSCR_S32G PB04_I2C1_SDA_CFG
+				    I2C1_SCL_IMCR PB03_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
 				    >;
 		};
 
-		pinctrl0_i2c1_gpio: pinctrl0_i2c1_gpio {
+		pinctrl_i2c1_gpio: pinctrl_i2c1_gpio {
 			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_GPIO
 				    PB04_MSCR_S32G PB04_I2C1_SDA_GPIO
+				    I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_i2c2: pinctrl0_i2c2 {
+		pinctrl_i2c2: pinctrl_i2c2 {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
 				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
+				    I2C2_SCL_IMCR PB05_I2C2_SCL_IN
+				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
 				    >;
 		};
 
-		pinctrl0_i2c2_gpio: pinctrl0_i2c2_gpio {
+		pinctrl_i2c2_gpio: pinctrl_i2c2_gpio {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
 				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
+				    I2C2_SCL_IMCR IMCR_DISABLED
+				    I2C2_SDA_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_i2c4: pinctrl0_i2c4 {
+		pinctrl_i2c4: pinctrl_i2c4 {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
 				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
+				    I2C4_SDA_IMCR PC01_I2C4_SDA_IN
+				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
 				    >;
 		};
 
-		pinctrl0_i2c4_gpio: pinctrl0_i2c4_gpio {
+		pinctrl_i2c4_gpio: pinctrl_i2c4_gpio {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
 				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
+				    I2C4_SDA_IMCR IMCR_DISABLED
+				    I2C4_SCL_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_qspi: pinctrl0_qspi {
+		pinctrl_qspi: pinctrl_qspi {
 			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
 				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
@@ -230,89 +242,30 @@
 				    >;
 		};
 
-		pinctrl0_dspi1: pinctrl0_dspi1 {
+		pinctrl_dspi1: pinctrl_dspi1 {
 			fsl,pins = <PA06_MSCR_S32G PA06_SPI1_SOUT_CFG
 				    PA07_MSCR_S32G PA07_SPI1_CS0_CFG
 				    PA08_MSCR_S32G PA08_SPI1_SCK_CFG
 				    PF15_MSCR_S32G PF15_SPI1_SIN_CFG
+				    DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
 				    >;
 		};
 
-		pinctrl0_dspi5: pinctrl0_dspi5 {
+		pinctrl_dspi5: pinctrl_dspi5 {
 			fsl,pins = <PA09_MSCR_S32G PA09_SPI5_SCK_CFG
 				    PA10_MSCR_S32G PA10_SPI5_SIN_CFG
 				    PA11_MSCR_S32G PA11_SPI5_SOUT_CFG
 				    PA12_MSCR_S32G PA12_SPI5_CS0_CFG
+				    DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
 				    >;
 		};
 
-		pinctrl0_usb: pinctrl0_usb {
+		pinctrl_usb: pinctrl_usb {
 			fsl,pins = <PD14_MSCR_S32G PD14_USB_DATA0_CFG
 				    PD15_MSCR_S32G PD15_USB_DATA1_CFG
 				    PE00_MSCR_S32G PE00_USB_DATA2_CFG
 				    PE01_MSCR_S32G PE01_USB_DATA3_CFG
-				    >;
-		};
-
-		pinctrl0_uart0: pinctrl0_uart0 {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
-				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
-				    LIN0_RX_IMCR PC10_LIN0_RX_IN>;
-		};
-	};
-};
-
-&pinctrl1 {
-	board_pinctrl1 {
-		pinctrl1_i2c1: pinctrl1_i2c1 {
-			fsl,pins = <I2C1_SCL_IMCR PB03_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
-			fsl,pins = <I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_i2c2: pinctrl1_i2c2 {
-			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
-				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c2_gpio: pinctrl1_i2c2_gpio {
-			fsl,pins = <I2C2_SCL_IMCR IMCR_DISABLED
-				    I2C2_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_i2c4: pinctrl1_i2c4 {
-			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
-				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
-				    >;
-		};
-
-		pinctrl1_i2c4_gpio: pinctrl1_i2c4_gpio {
-			fsl,pins = <I2C4_SDA_IMCR IMCR_DISABLED
-				    I2C4_SCL_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_dspi1: pinctrl1_dspi1 {
-			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
-				    >;
-		};
-
-		pinctrl1_dspi5: pinctrl1_dspi5 {
-			fsl,pins = <DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
-				    >;
-		};
-
-		pinctrl1_usb: pinctrl1_usb {
-			fsl,pins = <PH00_MSCR_S32G PH00_USB_DATA7_CFG
+				    PH00_MSCR_S32G PH00_USB_DATA7_CFG
 				    PL08_MSCR_S32G PL08_USB_CLK_CFG
 				    PL09_MSCR_S32G PL09_USB_DIR_CFG
 				    PL10_MSCR_S32G PL10_USB_STP_CFG
@@ -333,16 +286,23 @@
 				    USB_ULPI_NXT_IMCR PL11_USB_NXT_IN
 				    >;
 		};
+
+		pinctrl_uart0: pinctrl_uart0 {
+			u-boot,dm-pre-reloc;
+			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
+				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
+				    LIN0_RX_IMCR PC10_LIN0_RX_IN>;
+		};
 	};
 };
 
 &qspi {
-	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-0 = <&pinctrl_qspi>;
 	pinctrl-names = "default";
 };
 
 &uart0 {
-	pinctrl-0 = <&pinctrl0_uart0>;
+	pinctrl-0 = <&pinctrl_uart0>;
 	pinctrl-names = "default";
 	status = "okay";
 };
diff --git a/fdts/s32gxxxa-rdb.dtsi b/fdts/s32gxxxa-rdb.dtsi
index e228f1a06..d9e31c7ae 100644
--- a/fdts/s32gxxxa-rdb.dtsi
+++ b/fdts/s32gxxxa-rdb.dtsi
@@ -8,7 +8,7 @@
 
 &usdhc0 {
 	/* By default sd0 pins were able to work at 100Mhz and 200Mhz */
-	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-0 = <&pinctrl_sd0>;
 	pinctrl-1 = <>;
 	pinctrl-2 = <>;
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
@@ -16,13 +16,13 @@
 };
 
 &spi1 {
-	pinctrl-0 = <&pinctrl0_dspi1 &pinctrl1_dspi1>;
+	pinctrl-0 = <&pinctrl_dspi1>;
 	pinctrl-names = "default";
 	status = "okay";
 };
 
 &spi5 {
-	pinctrl-0 = <&pinctrl0_dspi5 &pinctrl1_dspi5>;
+	pinctrl-0 = <&pinctrl_dspi5>;
 	pinctrl-names = "default";
 	status = "okay";
 };
@@ -35,10 +35,10 @@
 &i2c0 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
-	scl-gpios = <&gpio0 32 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 31 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c0>;
+	pinctrl-1 = <&pinctrl_i2c0_gpio>;
+	scl-gpios = <&gpio 32 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 31 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -46,10 +46,10 @@
 &i2c1 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl1_i2c1>;
-	pinctrl-1 = <&pinctrl1_i2c1_gpio>;
-	scl-gpios = <&gpio1 163 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio1 165 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c1>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio 163 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 165 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -57,10 +57,10 @@
 &i2c2 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
-	pinctrl-1 = <&pinctrl0_i2c2_gpio &pinctrl1_i2c2_gpio>;
-	scl-gpios = <&gpio0 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c2>;
+	pinctrl-1 = <&pinctrl_i2c2_gpio>;
+	scl-gpios = <&gpio 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -68,10 +68,10 @@
 &i2c4 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
-	pinctrl-1 = <&pinctrl0_i2c4_gpio &pinctrl1_i2c4_gpio>;
-	scl-gpios = <&gpio0 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c4>;
+	pinctrl-1 = <&pinctrl_i2c4_gpio>;
+	scl-gpios = <&gpio 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 
 	vr5510@20 {
@@ -127,10 +127,10 @@
 	};
 };
 
-&pinctrl0 {
-	board_pinctrl0 {
+&pinctrl {
+	board_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_i2c0: pinctrl0_i2c0 {
+		pinctrl_i2c0: pinctrl_i2c0 {
 			fsl,pins = <PB15_MSCR_S32G PB15_I2C0_SDA_CFG
 				    PC00_MSCR_S32G PC00_I2C0_SCL_CFG
 				    I2C0_SDA_IMCR PB15_I2C0_SDA_IN
@@ -138,7 +138,7 @@
 				    >;
 		};
 
-		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+		pinctrl_i2c0_gpio: pinctrl_i2c0_gpio {
 			fsl,pins = <PB15_MSCR_S32G PB15_I2C0_SDA_GPIO
 				    PC00_MSCR_S32G PC00_I2C0_SCL_GPIO
 				    I2C0_SDA_IMCR IMCR_DISABLED
@@ -146,31 +146,55 @@
 				    >;
 		};
 
-		pinctrl0_i2c2: pinctrl0_i2c2 {
+		pinctrl_i2c1: pinctrl_i2c1 {
+			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_CFG
+				    PK05_MSCR_S32G PK05_I2C1_SDA_CFG
+				    I2C1_SCL_IMCR PK03_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PK05_I2C1_SDA_IN
+				    >;
+		};
+
+		pinctrl_i2c1_gpio: pinctrl_i2c1_gpio {
+			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_GPIO
+				    PK05_MSCR_S32G PK05_I2C1_SDA_GPIO
+				    I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
+		pinctrl_i2c2: pinctrl_i2c2 {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
 				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
+				    I2C2_SCL_IMCR PB05_I2C2_SCL_IN
+				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
 				    >;
 		};
 
-		pinctrl0_i2c2_gpio: pinctrl0_i2c2_gpio {
+		pinctrl_i2c2_gpio: pinctrl_i2c2_gpio {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
 				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
+				    I2C2_SCL_IMCR IMCR_DISABLED
+				    I2C2_SDA_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_i2c4: pinctrl0_i2c4 {
+		pinctrl_i2c4: pinctrl_i2c4 {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
 				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
+				    I2C4_SDA_IMCR PC01_I2C4_SDA_IN
+				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
 				    >;
 		};
 
-		pinctrl0_i2c4_gpio: pinctrl0_i2c4_gpio {
+		pinctrl_i2c4_gpio: pinctrl_i2c4_gpio {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
 				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
+				    I2C4_SDA_IMCR IMCR_DISABLED
+				    I2C4_SCL_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_qspi: pinctrl0_qspi {
+		pinctrl_qspi: pinctrl_qspi {
 			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
 				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
@@ -198,23 +222,25 @@
 				    >;
 		};
 
-		pinctrl0_dspi1: pinctrl0_dspi1 {
+		pinctrl_dspi1: pinctrl_dspi1 {
 			fsl,pins = <PA06_MSCR_S32G PA06_SPI1_SOUT_CFG
 				    PA07_MSCR_S32G PA07_SPI1_CS0_CFG
 				    PA08_MSCR_S32G PA08_SPI1_SCK_CFG
 				    PF15_MSCR_S32G PF15_SPI1_SIN_CFG
+				    DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
 				    >;
 		};
 
-		pinctrl0_dspi5: pinctrl0_dspi5 {
+		pinctrl_dspi5: pinctrl_dspi5 {
 			fsl,pins = <PA09_MSCR_S32G PA09_SPI5_SCK_CFG
 				    PA10_MSCR_S32G PA10_SPI5_SIN_CFG
 				    PA11_MSCR_S32G PA11_SPI5_SOUT_CFG
 				    PA12_MSCR_S32G PA12_SPI5_CS0_CFG
+				    DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
 				    >;
 		};
 
-		pinctrl0_uart0: pinctrl0_uart0 {
+		pinctrl_uart0: pinctrl_uart0 {
 			u-boot,dm-pre-reloc;
 			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
 				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
@@ -223,67 +249,13 @@
 	};
 };
 
-&pinctrl1 {
-	board_pinctrl1 {
-		pinctrl1_i2c1: pinctrl1_i2c1 {
-			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_CFG
-				    PK05_MSCR_S32G PK05_I2C1_SDA_CFG
-				    I2C1_SCL_IMCR PK03_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PK05_I2C1_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
-			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_GPIO
-				    PK05_MSCR_S32G PK05_I2C1_SDA_GPIO
-				    I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_i2c2: pinctrl1_i2c2 {
-			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
-				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c2_gpio: pinctrl1_i2c2_gpio {
-			fsl,pins = <I2C2_SCL_IMCR IMCR_DISABLED
-				    I2C2_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_i2c4: pinctrl1_i2c4 {
-			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
-				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
-				    >;
-		};
-
-		pinctrl1_i2c4_gpio: pinctrl1_i2c4_gpio {
-			fsl,pins = <I2C4_SDA_IMCR IMCR_DISABLED
-				    I2C4_SCL_IMCR IMCR_DISABLED
-				    >;
-		};
-
-		pinctrl1_dspi1: pinctrl1_dspi1 {
-			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
-				    >;
-		};
-
-		pinctrl1_dspi5: pinctrl1_dspi5 {
-			fsl,pins = <DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
-				    >;
-		};
-	};
-};
-
 &qspi {
-	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-0 = <&pinctrl_qspi>;
 	pinctrl-names = "default";
 };
 
 &uart0 {
-	pinctrl-0 = <&pinctrl0_uart0>;
+	pinctrl-0 = <&pinctrl_uart0>;
 	pinctrl-names = "default";
 	status = "okay";
 };
diff --git a/fdts/s32r45-evb.dts b/fdts/s32r45-evb.dts
index 903c50b0e..34735e909 100644
--- a/fdts/s32r45-evb.dts
+++ b/fdts/s32r45-evb.dts
@@ -13,7 +13,7 @@
 
 &qspi {
 	compatible = "nxp,s32r45-qspi", "nxp,s32cc-qspi";
-	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-0 = <&pinctrl_qspi>;
 	pinctrl-names = "default";
 	status = "okay";
 
@@ -27,7 +27,7 @@
 };
 
 &usdhc0 {
-	pinctrl-0 = <&pinctrl0_sd0>;
+	pinctrl-0 = <&pinctrl_sd0>;
 	pinctrl-names = "default";
 	no-1-8-v;
 	status = "okay";
@@ -40,8 +40,8 @@
 };
 
 &gmac0 {
-	pinctrl-0 = <&pinctrl0_gmac0 &pinctrl0_gmac0_mdio>;
-	pinctrl-1 = <&pinctrl0_gmac0_mdio>;
+	pinctrl-0 = <&pinctrl_gmac0 &pinctrl_gmac0_mdio>;
+	pinctrl-1 = <&pinctrl_gmac0_mdio>;
 	pinctrl-names = "gmac_rgmii", "gmac_sgmii";
 	status = "okay";
 	phy-mode = "rgmii";
@@ -61,10 +61,10 @@
 &i2c0 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
-	scl-gpios = <&gpio0 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c0>;
+	pinctrl-1 = <&pinctrl_i2c0_gpio>;
+	scl-gpios = <&gpio 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -72,51 +72,63 @@
 &i2c1 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
-	pinctrl-1 = <&pinctrl0_i2c1_gpio &pinctrl1_i2c1_gpio>;
-	scl-gpios = <&gpio0 14 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
-	sda-gpios = <&gpio0 15 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	pinctrl-0 = <&pinctrl_i2c1>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio 14 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio 15 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
-&pinctrl0 {
-	board_pinctrl0 {
+&pinctrl {
+	board_pinctrl {
 		u-boot,dm-pre-reloc;
-		pinctrl0_dspi1: pinctrl0_dspi1 {
+		pinctrl_dspi1: pinctrl_dspi1 {
 			fsl,pins = <PB04_MSCR_S32R45 PB04_SPI1_SCK_CFG
 				    PB05_MSCR_S32R45 PB05_SPI1_SIN_CFG
 				    PB06_MSCR_S32R45 PB06_SPI1_SOUT_CFG
 				    PB07_MSCR_S32R45 PB07_SPI1_CS0_CFG
 				    PB08_MSCR_S32R45 PB08_SPI1_CS1_CFG
+				    DSPI1_SIN_IMCR PB05_SPI1_SIN_IN
 				    >;
 		};
 
-		pinctrl0_dspi2: pinctrl0_dspi2 {
+		pinctrl_dspi2: pinctrl_dspi2 {
 			fsl,pins = <PB11_MSCR_S32R45 PB11_SPI2_SCK_CFG
 				    PB13_MSCR_S32R45 PB13_SPI2_SOUT_CFG
 				    PB14_MSCR_S32R45 PB14_SPI2_CS0_CFG
 				    PB15_MSCR_S32R45 PB15_SPI2_CS1_CFG
 				    PC01_MSCR_S32R45 PC01_SPI2_SIN_CFG
+				    DSPI2_SIN_IMCR PC01_SPI2_SIN_IN
 				    >;
 		};
 
-		pinctrl0_dspi3: pinctrl0_dspi3 {
+		pinctrl_dspi3: pinctrl_dspi3 {
 			fsl,pins = <PC04_MSCR_S32R45 PC04_SPI3_SCK_CFG
 				    PC06_MSCR_S32R45 PC06_SPI3_SIN_CFG
 				    PC07_MSCR_S32R45 PC07_SPI3_CS0_CFG
 				    PC08_MSCR_S32R45 PC08_SPI3_CS1_CFG
 				    PC13_MSCR_S32R45 PC13_SPI3_SOUT_CFG
+				    DSPI3_SIN_IMCR PC06_SPI3_SIN_IN
 				    >;
 		};
 
-		pinctrl0_gmac0_mdio: pinctrl0_gmac0_mdio {
+		pinctrl_dspi5: pinctrl_dspi5 {
+			fsl,pins = <PK00_MSCR_S32R45 PK00_SPI5_SCK_CFG
+				    PK03_MSCR_S32R45 PK03_SPI5_CS0_CFG
+				    PK04_MSCR_S32R45 PK04_SPI5_SIN_CFG
+				    PK05_MSCR_S32R45 PK05_SPI5_SOUT_CFG
+				    DSPI5_SIN_IMCR PK04_SPI5_SIN_IN
+				    >;
+		};
+
+		pinctrl_gmac0_mdio: pinctrl_gmac0_mdio {
 			fsl,pins = <PD12_MSCR_S32R45 PD12_GMAC0_MDC_CFG
 				    PD13_MSCR_S32R45 PD13_GMAC0_MDIO_CFG
 				    GMAC0_MDIO_IMCR PD13_GMAC0_MDIO_IN
 				    >;
 		};
 
-		pinctrl0_gmac0: pinctrl0_gmac0 {
+		pinctrl_gmac0: pinctrl_gmac0 {
 			fsl,pins = <PE02_MSCR_S32R45 PE02_GMAC0_TX_CLK_CFG
 				    PE03_MSCR_S32R45 PE03_GMAC0_TX_EN_CFG
 				    PE04_MSCR_S32R45 PE04_GMAC0_TX_D0_CFG
@@ -139,7 +151,7 @@
 				    >;
 		};
 
-		pinctrl0_i2c0: pinctrl0_i2c0 {
+		pinctrl_i2c0: pinctrl_i2c0 {
 			fsl,pins = <PB00_MSCR_S32R45 PB00_I2C0_SDA_CFG
 				    PB01_MSCR_S32R45 PB01_I2C0_SCL_CFG
 				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
@@ -147,7 +159,7 @@
 				    >;
 		};
 
-		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+		pinctrl_i2c0_gpio: pinctrl_i2c0_gpio {
 			fsl,pins = <PB00_MSCR_S32R45 PB00_I2C0_SDA_GPIO
 				    PB01_MSCR_S32R45 PB01_I2C0_SCL_GPIO
 				    I2C0_SDA_IMCR IMCR_DISABLED
@@ -155,19 +167,23 @@
 				    >;
 		};
 
-		pinctrl0_i2c1: pinctrl0_i2c1 {
+		pinctrl_i2c1: pinctrl_i2c1 {
 			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_CFG
 				    PA15_MSCR_S32R45 PA15_I2C1_SDA_CFG
+				    I2C1_SCL_IMCR PA14_I2C1_SCL_IN
+				    I2C1_SDA_IMCR PA15_I2C1_SDA_IN
 				    >;
 		};
 
-		pinctrl0_i2c1_gpio: pinctrl0_i2c1_gpio {
+		pinctrl_i2c1_gpio: pinctrl_i2c1_gpio {
 			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_GPIO
 				    PA15_MSCR_S32R45 PA15_I2C1_SDA_GPIO
+				    I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
 				    >;
 		};
 
-		pinctrl0_qspi: pinctrl0_qspi {
+		pinctrl_qspi: pinctrl_qspi {
 			fsl,pins = <PF05_MSCR_S32R45 PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32R45 PF06_QSPI_DATA_A1_CFG
 				    PF07_MSCR_S32R45 PF07_QSPI_DATA_A2_CFG
@@ -195,7 +211,7 @@
 				    >;
 		};
 
-		pinctrl0_sd0: pinctrl0_sd0 {
+		pinctrl_sd0: pinctrl_sd0 {
 			fsl,pins = <PC14_MSCR_S32R45 PC14_SD0_CLK_CFG
 				    PC15_MSCR_S32R45 PC15_SD0_CMD_CFG
 				    PD00_MSCR_S32R45 PD00_SD0_D0_CFG
@@ -220,7 +236,7 @@
 				    >;
 		};
 
-		pinctrl0_uart0: pinctrl0_uart0 {
+		pinctrl_uart0: pinctrl_uart0 {
 			u-boot,dm-pre-reloc;
 			fsl,pins = <PC09_MSCR_S32R45 PC09_LIN0_TX_CFG
 				    PC10_MSCR_S32R45 PC10_LIN0_RX_CFG
@@ -229,67 +245,27 @@
 	};
 };
 
-&pinctrl1 {
-	board_pinctrl1 {
-		pinctrl1_dspi1: pinctrl1_dspi1 {
-			fsl,pins = <DSPI1_SIN_IMCR PB05_SPI1_SIN_IN
-				    >;
-		};
-
-		pinctrl1_dspi2: pinctrl1_dspi2 {
-			fsl,pins = <DSPI2_SIN_IMCR PC01_SPI2_SIN_IN
-				    >;
-		};
-
-		pinctrl1_dspi3: pinctrl1_dspi3 {
-			fsl,pins = <DSPI3_SIN_IMCR PC06_SPI3_SIN_IN
-				    >;
-		};
-
-		pinctrl1_dspi5: pinctrl1_dspi5 {
-			fsl,pins = <PK00_MSCR_S32R45 PK00_SPI5_SCK_CFG
-				    PK03_MSCR_S32R45 PK03_SPI5_CS0_CFG
-				    PK04_MSCR_S32R45 PK04_SPI5_SIN_CFG
-				    PK05_MSCR_S32R45 PK05_SPI5_SOUT_CFG
-				    DSPI5_SIN_IMCR PK04_SPI5_SIN_IN
-				    >;
-		};
-
-		pinctrl1_i2c1: pinctrl1_i2c1 {
-			fsl,pins = <I2C1_SCL_IMCR PA14_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PA15_I2C1_SDA_IN
-				    >;
-		};
-
-		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
-			fsl,pins = <I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
-		};
-	};
-};
-
 &spi1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl0_dspi1>, <&pinctrl1_dspi1>;
+	pinctrl-0 = <&pinctrl_dspi1>;
 	status = "okay";
 };
 
 &spi2 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl0_dspi2>, <&pinctrl1_dspi2>;
+	pinctrl-0 = <&pinctrl_dspi2>;
 	status = "okay";
 };
 
 &spi3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl0_dspi3>, <&pinctrl1_dspi3>;
+	pinctrl-0 = <&pinctrl_dspi3>;
 	status = "okay";
 };
 
 &spi5 {
 	pinctrl-names = "default";
-	pinctrl-0 =  <&pinctrl1_dspi5>;
+	pinctrl-0 =  <&pinctrl_dspi5>;
 	status = "okay";
 };
 
@@ -298,7 +274,7 @@
 };
 
 &uart0 {
-	pinctrl-0 = <&pinctrl0_uart0>;
+	pinctrl-0 = <&pinctrl_uart0>;
 	pinctrl-names = "default";
 	status = "okay";
 };
diff --git a/fdts/s32r45.dtsi b/fdts/s32r45.dtsi
index 66d655be7..985f02c37 100644
--- a/fdts/s32r45.dtsi
+++ b/fdts/s32r45.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/nvmem/s32cc-siul2-nvmem.h>
 #include <dt-bindings/pinctrl/s32r45-pinctrl.h>
+#include <dt-bindings/memory/s32-siul2.h>
 
 #include "s32cc.dtsi"
 / {
@@ -70,31 +71,85 @@
 	};
 
 	soc {
-		siul2_0@4009c000 {
+		siul2@4009c000 {
 			compatible = "simple-mfd";
 			status = "okay";
-			reg = <0x0 0x4009c000 0x0 0x2000>;
 			u-boot,dm-pre-reloc;
-
-			pinctrl0: siul2-pinctrl0 {
-				compatible = "nxp,s32cc-siul2-pinctrl";
+			#address-cells = <2>;
+			#size-cells = <2>;
+
+				 /* MIDR */
+			ranges = <MIDR_SIUL2_0 0 0x0 0x4009c000 0x0 0x10>,
+				 <MIDR_SIUL2_1 0 0x0 0x4403c000 0x0 0x10>,
+
+				 /* MSCR 0-101 */
+				 <MSCRS_SIUL2_0 0 0x0 0x4009c240 0x0 0x198>,
+				 /* MSCR 102-133 */
+				 <MSCRS_SIUL2_1 0 0x0 0x4403c3d8 0x0 0x80>,
+
+				 /* IMCR range 512–595 */
+				 <IMCRS_SIUL2_0 0 0x0 0x4009ca40 0x0 0x150>,
+				 /* IMCR range 603 - 695 */
+				 <IMCRS_SIUL2_1_0 0 0x0 0x4403cbac 0x0 0x174>,
+				 /* IMCR range 696 - 727 (EIRQs) */
+				 <IMCRS_SIUL2_1_EIRQS 0 0x0 0x4403cd20 0x0 0x80>,
+				 /* IMCR range 728 - 785 */
+				 <IMCRS_SIUL2_1_1 0 0x0 0x4403cda0 0x0 0xe8>,
+
+				 /* Output pads */
+				 <OPADS_SIUL2_0 0 0x0 0x4009d700 0x0 0x10>,
+				 <OPADS_SIUL2_1 0 0x0 0x4403d700 0x0 0x14>,
+
+				 /* Input pads */
+				 <IPADS_SIUL2_0 0 0x0 0x4009d740 0x0 0x10>,
+				 <IPADS_SIUL2_1 0 0x0 0x4403d740 0x0 0x14>,
+
+				/* EIRQs */
+				 <EIRQS_SIUL2_1 0 0x0 0x4403c010 0x0 0xb4>;
+
+			pinctrl: siul2-pinctrl@4009c240 {
+				compatible = "nxp,s32r45-siul2-pinctrl";
 				#pinctrl-cells = <2>;
+
+				reg = <MSCRS_SIUL2_0   0 0x0 0x198>,
+				      <MSCRS_SIUL2_1   0 0x0 0x80>,
+				      <IMCRS_SIUL2_0   0 0x0 0x150>,
+				      <IMCRS_SIUL2_1_0 0 0x0 0x174>,
+				      <IMCRS_SIUL2_1_1 0 0x0 0xe8>;
+
 				/* MSCR range */
-				pins = <&pinctrl0 0 101>,
+				nxp,pins = <0   101
+					    102 133
 				/* IMCR range */
-				<&pinctrl0 512 573>;
+					    512 595
+					    603 695
+					    728 785>;
 				status = "okay";
 			};
 
-			gpio0: siul2-gpio0 {
+			gpio: siul2-gpio@4009d700 {
 				compatible = "nxp,s32cc-siul2-gpio";
+
+				reg = <OPADS_SIUL2_0       0 0x0 0x10>,
+				      <OPADS_SIUL2_1       0 0x0 0x14>,
+				      <IPADS_SIUL2_0       0 0x0 0x10>,
+				      <IPADS_SIUL2_1       0 0x0 0x14>,
+				      <EIRQS_SIUL2_1       0 0x0 0xb4>,
+				      <IMCRS_SIUL2_1_EIRQS 0 0x0 0x80>;
+				reg-names = "opads0", "opads1", "ipads0",
+					    "ipads1", "eirqs", "eirq-imcrs";
+
 				#gpio-cells = <2>;
 				gpio-controller;
-				gpio-ranges = <&pinctrl0 0 0 102>;
+					/* GPIO 0-101 */
+				gpio-ranges = <&pinctrl 0 0 102>,
+					/* GPIO 102-133 */
+					      <&pinctrl 102 102 32>;
 				status = "okay";
 			};
 
 			siul2_0_nvram: siul2_0_nvram {
+				reg = <MIDR_SIUL2_0 0 0x0 0x10>;
 				#address-cells = <1>;
 				#size-cells = <1>;
 
@@ -144,6 +199,29 @@
 				};
 
 			};
+
+			siul2_1_nvram {
+				reg = <MIDR_SIUL2_1 0 0x0 0x10>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				compatible = "nxp,s32r-siul2_1-nvram";
+				status = "okay";
+				u-boot,dm-pre-reloc;
+
+				serdes_presence: serdes_presence@100 {
+					reg = <S32CC_SERDES_PRESENCE S32CC_CELL_SIZE>;
+				};
+
+				lax_presence: lax_presence@104 {
+					reg = <S32CC_LAX_PRESENCE S32CC_CELL_SIZE>;
+				};
+
+				soc_subminor: soc_subminor@108 {
+					reg = <S32CC_SOC_SUBMINOR S32CC_CELL_SIZE>;
+					u-boot,dm-pre-reloc;
+				};
+			};
 		};
 
 		gmac1: ethernet@44010000 {
@@ -183,54 +261,6 @@
 			};
 		};
 
-		siul2_1@4403c000 {
-			compatible = "simple-mfd";
-			status = "okay";
-			reg = <0x0 0x4403c000 0x0 0x2000>;
-			u-boot,dm-pre-reloc;
-
-			pinctrl1: siul2-pinctrl1 {
-				compatible = "nxp,s32cc-siul2-pinctrl";
-				#pinctrl-cells = <2>;
-				/* MSCR range */
-				pins = <&pinctrl1 102 133>,
-				/* IMCR range */
-				<&pinctrl1 603 785>;
-				status = "okay";
-			};
-
-			gpio1: siul2-gpio1 {
-				compatible = "nxp,s32cc-siul2-gpio";
-				gpio-controller;
-				#gpio-cells = <2>;
-				gpio-ranges = <&pinctrl1 0 102 32>;
-				status = "okay";
-			};
-
-			siul2_1_nvram {
-				#address-cells = <1>;
-				#size-cells = <1>;
-
-				compatible = "nxp,s32r-siul2_1-nvram";
-				status = "okay";
-				u-boot,dm-pre-reloc;
-
-				serdes_presence: serdes_presence@100 {
-					reg = <S32CC_SERDES_PRESENCE S32CC_CELL_SIZE>;
-				};
-
-				lax_presence: lax_presence@104 {
-					reg = <S32CC_LAX_PRESENCE S32CC_CELL_SIZE>;
-				};
-
-				soc_subminor: soc_subminor@108 {
-					reg = <S32CC_SOC_SUBMINOR S32CC_CELL_SIZE>;
-					u-boot,dm-pre-reloc;
-				};
-
-			};
-		};
-
 		mc_cgm2: mc_cgm2@440c0000 {
 			compatible = "nxp,s32cc-mc_cgm2";
 			reg = <0x0 0x440c0000 0x0 0x3000>;
-- 
2.17.1

